keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -0,0 +1,119 @@
+             documentation.  Defaults to `"bilinear"`.
+        fill_mode: fill_mode in the `ImageProjectiveTransformV3` op.
+             Supported values can be found in the `ImageProjectiveTransformV3`
+             documentation.  Defaults to `"reflect"`.
+        fill_value: fill_value in the `ImageProjectiveTransformV3` op.
+             Supported values can be found in the `ImageProjectiveTransformV3`
+             documentation.  Defaults to `0.0`.
+    """
+
+    def __init__(
+        self,
+        x=None,
+        y=None,
+        interpolation="bilinear",
+        fill_mode="reflect",
+        fill_value=0.0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if isinstance(x, float):
+            x = (0, x)
+        if isinstance(y, float):
+            y = (0, y)
+        if x is None and y is None:
+            warnings.warn(
+                "RandomShear received both `x=None` and `y=None`.  As a "
+                "result, the layer will perform no augmentation."
+            )
+        self.x = x
+        self.y = y
+        self.interpolation = interpolation
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+
+    def get_random_tranformation(self):
+        x = self._get_shear_amount(self.x)
+        y = self._get_shear_amount(self.y)
+        return (x, y)
+
+    def _get_shear_amount(self, constraint):
+        if constraint is None:
+            return None
+
+        negate = self._random_generator.random_uniform((), 0, 1, dtype=tf.float32) > 0.5
+        negate = tf.cond(negate, lambda: -1.0, lambda: 1.0)
+
+        return negate * self._random_generator.random_uniform(
+            (), constraint[0], constraint[1]
+        )
+
+    def augment_image(self, image, transformation=None):
+        image = tf.expand_dims(image, axis=0)
+
+        x, y = transformation
+
+        if x is not None:
+            transform_x = RandomShear._format_transform(
+                [1.0, x, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0]
+            )
+            image = preprocessing.transform(
+                images=image,
+                transforms=transform_x,
+                interpolation=self.interpolation,
+                fill_mode=self.fill_mode,
+                fill_value=self.fill_value,
+            )
+
+        if y is not None:
+            transform_y = RandomShear._format_transform(
+                [1.0, 0.0, 0.0, y, 1.0, 0.0, 0.0, 0.0]
+            )
+            image = preprocessing.transform(
+                images=image,
+                transforms=transform_y,
+                interpolation=self.interpolation,
+                fill_mode=self.fill_mode,
+                fill_value=self.fill_value,
+            )
+
+        return tf.squeeze(image, axis=0)
+
+    @staticmethod
+    def _format_transform(transform):
+        transform = tf.convert_to_tensor(transform, dtype=tf.float32)
+        return transform[None]
```

Comment:
Humm, what is transform[None]?

Reply:
it adds a new axis on [1,].  Its required to use `transform` as ImageProjectiveTransform expects a batch dim.

Reply:
https://stackoverflow.com/questions/1408311/numpy-array-slice-using-none same is supported in TensorFlow - using `None` is equivalent to `tf.new_axis`

Reply:
Updated to `tf.newaxis` for readability.